### PR TITLE
added a payment page

### PR DESCRIPTION
**Pull request by anjalisoni3655**
Made a payment page that will show  the following details:
- Username
- Address
- Offers availed
- Policy purchased
- Initial cost of the policy
- Total discount
- Final cost
At the end, it will have a button to confirm payment and cancel payment button.
pay

### CODE REVIEW

**Comment by oshhh** [comment]
Please add a screenshot to make the review easier,

File: home_insurance_app/.gitignore [hunk 1]
```diff
@@ -43,4 +43,4 @@ app.*.map.json
 !/packages/flutter_tools/test/data/dart_dependencies_test/**/.packages
 home_insurance_app/android/app/src/main/res/*
 home_insurance_app/android/app/google-service.json
-
+google-services.json
```
Comment (dreamplay) [inline home_insurance_app/.gitignore:46]: what exactly is in this?

File: home_insurance_app/lib/pages/payment_page.dart [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      purchase = {
+//        'structure_id': arguments['structure']['id'],
+        'address': arguments['userAddress'],
+        'policy': arguments['selectedPolicy'],
+        'offer': arguments['selectedOffer'],
+        'total_discount': arguments['selectedPolicy'].cost *
+            0.01 *
+            arguments['selectedOffer'].discount,
+        'discounted_cost': arguments['selectedPolicy'].cost *
+            (1 - 0.01 * arguments['selectedOffer'].discount)
```
Comment (dreamplay) [inline home_insurance_app/lib/pages/payment_page.dart:27]: please extract this discounting formula to a utility method

File: home_insurance_app/lib/pages/payment_page.dart [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      };
+      print(purchase);
+    }
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Payment details'),
+        centerTitle: true,
+        backgroundColor: Colors.brown,
+      ),
```
Comment (dreamplay) [inline home_insurance_app/lib/pages/payment_page.dart:37]: add a TODO to extract this CSS / layout details to separate file/method

File: home_insurance_app/lib/pages/payment_page.dart [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+                leftText: 'Discounted Cost: ',
+                rightText: 'Rs ${purchase['discounted_cost']}' ?? '',
+              ),
+            ],
+          ),
+        ),
+      ),
+      floatingActionButton: FloatingActionButton.extended(
+        onPressed: () {
+          addInsurancePurchased(purchase);
```
Comment (dreamplay) [inline home_insurance_app/lib/pages/payment_page.dart:99]: do we not have a back or cancel button from payments page?

File: home_insurance_app/lib/pages/show_discounts.dart [hunk 5]
```diff
@@ -89,15 +90,17 @@ class DisplayDiscountsState extends State<DisplayDiscounts> {
                                   CustomTextStyle(fontWeight: FontWeight.w900),
                             ),
                             onPressed: () {
+                              print(data['selectedPolicy']);
```
Comment (dreamplay) [inline home_insurance_app/lib/pages/show_discounts.dart:93]: remove print

File: home_insurance_app/lib/data/database_utils.dart [hunk 6]
```diff
@@ -0,0 +1,5 @@
+void addInsurancePurchased(purchase) {
+  print("added");
```
Comment (dreamplay) [inline home_insurance_app/lib/data/database_utils.dart:2]: Why do we have print statements for master branch?

File: home_insurance_app/test/payment_test.dart [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:homeinsuranceapp/data/offer.dart';
+import 'package:homeinsuranceapp/data/policy.dart';
+import 'package:homeinsuranceapp/data/user_home_details.dart';
+import 'package:homeinsuranceapp/pages/payment_page.dart';
+
+void main() {
```
Comment (dreamplay) [inline home_insurance_app/test/payment_test.dart:8]: pls add proper spacing between tests and declarations

File: home_insurance_app/.gitignore (hunk 1 above)
Comment (anjalisoni3655) [reply home_insurance_app/.gitignore:46]: When we register the project we get this json file that uniquely identifies our project and we can use the firebase features like sign in, storage, push notification, remote config.

File: home_insurance_app/lib/data/database_utils.dart (hunk 6 above)
Comment (anjalisoni3655) [reply home_insurance_app/lib/data/database_utils.dart:2]: I will remove all of them, that was for debugging.

File: home_insurance_app/lib/pages/payment_page.dart (hunk 4 above)
Comment (anjalisoni3655) [reply home_insurance_app/lib/pages/payment_page.dart:99]: We have the back arrow in the app bar right now. Should I make a cancel button as well that will send the user back to the home screen?

File: home_insurance_app/lib/pages/payment_page.dart (hunk 2 above)
Comment (anjalisoni3655) [reply home_insurance_app/lib/pages/payment_page.dart:27]: yah will do that

Comment (oshhh) [reply home_insurance_app/lib/pages/payment_page.dart:27]: Done that in the firebase purchase branch. Removed the json added an object of Purchase class instead.

File: home_insurance_app/lib/data/database_utils.dart [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+  print(purchase);
+  // TODO: push purchase to database
+}
+
+double getFinalDiscount({int cost, int discount}) {
+  return cost * 0.01 * discount;
+}
+
+double getFinalAmount({int cost, int discount}) {
+  return cost * (1 - 0.01 * discount);
```
[the remark this answers is not in the document]
Comment (anjalisoni3655) [reply home_insurance_app/lib/data/database_utils.dart:11]: thanks, will do that

File: home_insurance_app/pubspec.lock [hunk 9]
```diff
@@ -14,35 +14,35 @@ packages:
       name: archive
       url: "https://pub.dartlang.org"
```
[the remark this answers is not in the document]
Comment (anjalisoni3655) [reply home_insurance_app/pubspec.lock]: tested, It's working.

**Pull request merged**